### PR TITLE
Adding section on non-Cartesian datasets

### PR DESCRIPTION
**Pull request by matthewturk**
Still needs an image of the pixelized results.

### CODE REVIEW

**Comment by matthewturk** [comment]
Waiting on @neutrinoceros to make sure the citation on the image is right

File: scripts/spherical_data_figure.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import yt
+import yt_idefix
```
Comment (neutrinoceros) [inline scripts/spherical_data_figure.py:2]: this line is not strictly needed anymore, I would suggest replacing it with a comment to point out that `yt-idefix` is required to run the script.

File: content/30.abstracting_simulation_types.md [hunk 2]
```diff
@@ -283,3 +283,10 @@ Despite these weak spots, however, `yt` does provide a number of routines that a
 (See @sec:pixelization for more detail on this process.)
 Pixelizers that take variable-resolution data along the $r$ and $\theta$ axes have been made available (for slicing along a conical section of a sphere or along the $z$ axis of a cylinder) as well as very simple projections from the surface of a sphere to a flat image (specifically utilizing the Aitoff projection).
 `yt` also provides access to Cartopy [@doi:10.5281/zenodo.7430317] for more advanced or featureful map projections, as well as overlaying continents and other geographic shapes.
+In Figure @fig:spherical_data we demonstrate some of the native, built-in functionality `yt` provides for non-cartesian data.
+This dataset, a simulation of MHD-driven winds in a protoplanetary disk, was conducted in spherical coordinates.
```
Comment (neutrinoceros) [inline content/30.abstracting_simulation_types.md:287]:
```suggestion
This dataset, a simulation of magnetically-driven winds in a protoplanetary disk, was conducted in spherical coordinates.
```

File: content/30.abstracting_simulation_types.md [hunk 3]
```diff
@@ -283,3 +283,10 @@ Despite these weak spots, however, `yt` does provide a number of routines that a
...
+Here, we have used `yt`'s functionality for overplotting streamlines as well as line integral convolution on irregular meshes to display the data in its native resolution and as a slice along the poloidal axis of the simulation domain.
```
Comment (neutrinoceros) [inline content/30.abstracting_simulation_types.md:288]: "in the poloidal plane" or "along the azimuth axis" are equally valid, but "along the poloidal axis" isn't
```suggestion
Here, we have used `yt`'s functionality for overplotting streamlines as well as line integral convolution on irregular meshes to display the data in its native resolution and as a slice along the azimuthal axis of the simulation domain.
```

**Comment by neutrinoceros** [comment]
We should cite this paper https://ui.adsabs.harvard.edu/abs/2021A%26A...647A.192J/abstract for the original simulation

File: scripts/spherical_data_figure.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    "data.0092.vtk",
+    units_override={
+        "length_unit": (100, 'AU'),
+        "mass_unit": (1, "Msun"),
+    }
+)
+p = yt.SlicePlot(ds, "phi", "density")
+p.annotate_streamlines(
+    ("gas", "velocity_cartesian_x"),
+    ("gas", "velocity_cartesian_z"),
```
Comment (neutrinoceros) [inline scripts/spherical_data_figure.py:14]:
```suggestion
    ("gas", "velocity_cylindrical_radius"),
    ("gas", "velocity_cylindrical_z"),
```

File: scripts/spherical_data_figure.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    color=("gas", "velocity_magnitude"),
+    cmap="turbo",
+    density=0.8,
+    broken_streamlines=False,
+)
+p.annotate_line_integral_convolution(
+    "magnetic_field_cartesian_x",
+    "magnetic_field_cartesian_z",
+    alpha=0.35,
+)
```
Comment (neutrinoceros) [inline scripts/spherical_data_figure.py:24]: Some aesthetic changes + an actual fix for streamlines
```suggestion
p = yt.SlicePlot(ds, "phi", "density")
p.annotate_streamlines(
    ("gas", "velocity_cylindrical_radius"),
    ("gas", "velocity_cylindrical_z"),
    color=("gas", "velocity_magnitude"),
    cmap="Greys_r",
    density=1,
    arrowsize=0.5,
    linewidth=0.7,
    broken_streamlines=False,
)
p.annotate_line_integral_convolution(
    "magnetic_field_cartesian_x",
    "magnetic_field_cartesian_z",
    alpha=0.3,
)
```

**Comment by neutrinoceros** [comment]
@matthewturk did you forget to include the citation ?

**Pull request merged**